xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionsDataConsumer.java

```diff
@@ -130,26 +130,29 @@ public void consume(MentionsData data) throws XWikiException
             this.initContext(author, data.getWikiId());
             DocumentReference dr = this.documentReferenceResolver.resolve(data.getDocumentReference());
             XWikiDocument doc = this.documentRevisionProvider.getRevision(dr, data.getVersion());
-            Syntax syntax = doc.getSyntax();
-            XWikiDocument oldDoc;
-            if (doc.getPreviousVersion() != null) {
-                oldDoc = this.documentRevisionProvider.getRevision(dr, doc.getPreviousVersion());
-            } else {
-                oldDoc = null;
-            }
-            DocumentReference documentReference = doc.getDocumentReference();
-            DocumentReference authorReference = doc.getAuthorReference();
-
-            if (oldDoc == null) {
-                // CREATE
-                handleOnCreate(doc.getXDOM(), documentReference, authorReference, MentionLocation.DOCUMENT);
-                traverseXObjectsOnCreate(doc.getXObjects(), documentReference, authorReference, syntax);
-            } else {
-                // UPDATE
-                handleOnUpdate(oldDoc.getXDOM(), doc.getXDOM(), documentReference, authorReference,
-                    MentionLocation.DOCUMENT);
-                traverseXObjectsOnUpdate(oldDoc.getXObjects(), doc.getXObjects(), documentReference, authorReference,
-                    syntax);
+            if (doc != null) {
+                Syntax syntax = doc.getSyntax();
+                XWikiDocument oldDoc;
+                if (doc.getPreviousVersion() != null) {
+                    oldDoc = this.documentRevisionProvider.getRevision(dr, doc.getPreviousVersion());
+                } else {
+                    oldDoc = null;
+                }
+                DocumentReference documentReference = doc.getDocumentReference();
+                DocumentReference authorReference = doc.getAuthorReference();
+
+                if (oldDoc == null) {
```

Comment:
this if should be mergeable with the previous IF, no?

Reply:
It is. I've simplified the method.